tests/lib/external/snapd-testing-tools/remote/remote.refresh

```diff
@@ -0,0 +1,258 @@
+#!/bin/bash -e
+
+show_help() {
+    echo "usage: remote.refresh snap [--channel CHANNEL] <SNAPNAME>"
+    echo "usage: remote.refresh full [--channel CHANNEL]"
+    echo ""
+    echo "SNAPNAME: allowed options are: kernel, core, base, gadget, snapd or the snap name"
+    echo ""
+    echo "Available options:"
+    echo "  -h --help   show this help message."
+    echo ""
+}
+
+check_refresh_after_reboot() {
+    local refresh_channel=$1
+    local snap=$2
+
+    remote.wait-for no-ssh -n 60 --wait 2
+    remote.wait-for ssh -n 120 --wait 2
+    remote.wait-for snap-command
+    remote.retry -n 10 --wait 2  "snap info $snap | grep -q -E \"tracking: +(latest/${refresh_channel}|${refresh_channel})\""
+}
+
+check_refresh() {
+    local refresh_channel=$1
+    local snap=$2
+
+    remote.wait-for ssh -n 60 --wait 2
+    remote.wait-for snap-command
+    remote.retry -n 10 --wait 2  "snap info $snap | grep -q -E \"tracking: +(latest/${refresh_channel}|${refresh_channel})\""
+}
+
+do_refresh() {
+    local snap_name=$1
+    local refresh_channel=$2
+
+    output=$(remote.exec "sudo snap refresh --channel ${refresh_channel} $snap_name 2>&1" || echo "snapd is about to reboot the system")
+    if echo "$output" | grep -E "(no updates available|cannot refresh \"$snap_name\"|is not installed)"; then
+        echo "remote.refresh: snap \"$snap_name\" has no updates available"
+    elif echo "$output" | grep -E "snapd is about to reboot the system"; then
+        remote.exec --timeout 3 "sudo reboot" || true
+        check_refresh_after_reboot "$refresh_channel" "$snap_name"
+    else
+        check_refresh "$refresh_channel" "$snap_name"
+    fi
+}
+
+process_refresh() {
+    local snap_name=$1
+    local refresh_channel=$2
+    
+    if [ -z "$refresh_channel" ]; then
+        # Tracking is retrieved from snap info command because in old versions of
+        # snapd the tracking is not included in the snap list output
+        refresh_channel="$(remote.exec "snap info $snap_name" | grep -E '^tracking:' |  awk '{ print $2 }')"
+    fi
+
+    do_refresh "$snap_name" "$refresh_channel"
+}
+
+refresh_kernel() {
+    local refresh_channel=$1
+    local snap_name
+
+    check_ubuntu_core
+    snap_name="$(remote.exec "snap list" | grep -E '(kernel$|kernel,)' | awk '{ print $1 }')"
+    if [ -z "$snap_name" ]; then
+        echo "remote.refresh: no kernel snap to update"
+        return
+    fi
+
+    process_refresh "$snap_name" "$refresh_channel"
+}
```

Comment:
There's a lot of common code between this function and the ones below. I think we should try to make them call a common helper. Like this:
```suggestion
refresh_kernel() {
    check_ubuntu_core
    refresh_snap "$1"
}
```